src/components/search/SearchContext.jsx

```diff
@@ -1,12 +1,68 @@
-import React, { createContext, useState } from 'react';
+import React, {
+  createContext, useReducer, useMemo, useEffect,
+} from 'react';
 import PropTypes from 'prop-types';
+import { useLocation, useHistory } from 'react-router-dom';
+import qs from 'query-string';
+import {
+  BOOLEAN_FILTERS,
+  SEARCH_FACET_FILTERS,
+} from './data/constants';
+import { refinementsReducer } from './data/reducer';
+import { setMultipleRefinementsAction } from './data/actions';
+import { updateRefinementsFromQueryParams } from './data/utils';
+import { useIsFirstRender } from '../utils/hooks';
 
 export const SearchContext = createContext();
 
 const SearchData = ({ children }) => {
-  // NOTE: If this context starts to store a lot of state, switch to useReducer
-  const [showAllCatalogs, setShowAllCatalogs] = useState(false);
-  const value = { showAllCatalogs, setShowAllCatalogs };
+  const [refinementsFromQueryParams, dispatch] = useReducer(refinementsReducer, {});
+
+  const { search } = useLocation();
+  const history = useHistory();
+
+  const queryParams = useMemo(
+    () => qs.parse(search),
+    [search],
+  );
+
+  useEffect(
+    () => {
+      const activeFacetAttributes = SEARCH_FACET_FILTERS.map(filter => filter.attribute);
+      const keysToSet = {};
+      Object.entries(queryParams).forEach(([key, value]) => {
+        if (activeFacetAttributes.includes(key)) {
+          const valueAsArray = value.includes(',') ? value.split(',') : [value];
+          keysToSet[key] = valueAsArray;
+        } else if (BOOLEAN_FILTERS.includes(key)) {
+          keysToSet[key] = +value;
```

Comment:
A comment explaining what the `+` is doing here might be helpful for our future selves.